lib/modules/manager/gitlabci-include/extract.ts

```diff
@@ -33,6 +42,57 @@ function extractDepFromIncludeFile(
   return dep;
 }
 
+function extractDepFromIncludeComponent(
+  includeComponent: GitlabIncludeComponent,
+  endpoint: string | undefined,
+): PackageDependency | null {
+  const componentReferenceMatch = componentReferenceRegex?.exec(
+    includeComponent.component,
+  );
+  if (!componentReferenceMatch?.groups) {
+    logger.debug(
+      { componentReference: includeComponent.component },
+      'Ignoring malformed component reference',
+    );
+    return null;
+  }
+  const projectPathParts =
+    componentReferenceMatch.groups.projectPath.split('/');
+  if (projectPathParts.length < 2) {
+    logger.debug(
+      { componentReference: includeComponent.component },
+      'Ignoring component reference with incomplete project path',
+    );
+    return null;
+  }
+
+  const dep: PackageDependency = {
+    datasource: GitlabTagsDatasource.id,
+    depName: componentReferenceMatch.groups.projectPath,
+    depType: 'repository',
+    currentValue: componentReferenceMatch.groups.specificVersion,
+  };
+  if (dep.currentValue === componentReferenceLatestVersion) {
+    logger.debug(
+      { componentVersion: dep.currentValue },
+      'Ignoring component version',
+    );
+    dep.skipReason = 'unsupported-version';
+  }
```

Comment:
`~latest` is a special syntax to indicate the most recent release. I think we can ignore it like for other managers.